### ISSUE TITLE
Access control should be implemented in compiler layer, not workspaces layer

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#24518]
See https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions_Accessibility.cs

These language questions (of accessibility) should be implemented in the compiler layer, not the workspaces layer.

See https://github.com/dotnet/roslyn/blob/94ab292d2c79b1675de0d6ff0a7361ecfbf20cfb/src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs for the C# implementation.

This is related to https://github.com/dotnet/roslyn/issues/23842, which requests these APIs be publicly available.

**Comment by CyrusNajmabadi** [comment]
Oh yes please.  This was a request a long time ago, but it was punted on and pushed on the IDE to do :)  I would much prefer the compiler provide the APIs to answer how to do this (especially as it's very likely the IDE code is wrong for some subtle case, or is not correct across both languages).

**Comment by gafter** [comment]
A new API added in https://github.com/dotnet/roslyn/pull/27312 may be useful to replace some or all uses of the IDE accessibility API.
